eduhelx_jupyterlab_prof/handlers.py

```diff
@@ -211,118 +130,259 @@ async def get_value(self, current_path: str):
-        value["assignments"] = assignments
 
-        current_assignment = student_repo.current_assignment
-        # The student is in their repo, but we still need to check if they're actually in an assignment directory.
-        if current_assignment is None:
-            # If user is not in an assignment, we're done. Just leave current_assignment as None.
-            return json.dumps(value)
-        
-        submissions = await self.api.get_my_submissions(current_assignment["id"])
-        for submission in submissions:
-            submission["commit"] = get_commit_info(submission["commit_id"], path=student_repo.repo_root)
-        current_assignment["submissions"] = submissions
-        current_assignment["staged_changes"] = []
-        for modified_path in get_modified_paths(path=student_repo.repo_root):
-            full_modified_path = Path(student_repo.repo_root) / modified_path["path"]
-            abs_assn_path = Path(student_repo.repo_root) / assignment["directory_path"]
-            try:
-                path_relative_to_assn = full_modified_path.relative_to(abs_assn_path)
-                modified_path["path_from_repo"] = modified_path["path"]
-                modified_path["path_from_assn"] = str(path_relative_to_assn)
-                current_assignment["staged_changes"].append(modified_path)
-            except ValueError:
-                # This path is not part of the current assignment directory
-                pass
+            assignment["staged_changes"] = []
+            for modified_path in get_modified_paths(path=student_repo.repo_root):
+                full_modified_path = Path(student_repo.repo_root) / modified_path["path"]
+                abs_assn_path = Path(student_repo.repo_root) / assignment["directory_path"]
+                try:
+                    path_relative_to_assn = full_modified_path.relative_to(abs_assn_path)
+                    modified_path["path_from_repo"] = modified_path["path"]
+                    modified_path["path_from_assn"] = str(path_relative_to_assn)
+                    assignment["staged_changes"].append(modified_path)
+                except ValueError:
+                    # This path is not part of the assignment directory
+                    pass
+
+        value["assignments"] = assignments
         
+        current_assignment = student_repo.current_assignment
+        if current_assignment:
+            current_assignment["student_submissions"] = await self.api.get_submissions(current_assignment["id"])
+            for student in current_assignment["student_submissions"]:
+                for submission in current_assignment["student_submissions"][student]:
+                    submission["commit"] = {
+                        "id": submission["commit_id"],
+                        "message": "",
+                        "author_name": "",
+                        "author_email": "",
+                        "committer_name": "",
```

Comment:
would the author name/email be different than the commiter name/email?

Reply:
No, they're the same in our ecosystem. It's just that git has both so we have both.